### PR TITLE
(RE-6503) Add wily cow

### PR DESCRIPTION
**Pull request by stahnma**

### CODE REVIEW

**Comment by melissa** [comment]
You need to edit `templates/pbuilderrc.erb` and you might also need to add a wily file similar to the stuff in https://github.com/puppetlabs/puppetlabs-debbuilder/tree/master/files

**Comment by melissa** [comment]
@stahnma do you want me to review this? Do you want me to finish making the changes? What are you asking of me here?

File: manifests/setup/cows.pp [hunk 1]
```diff
@@ -118,7 +121,7 @@
   # cows of. All Ubuntu varieties are symlinks pointing to `squeeze` and all debian
   # varieties are symlinks pointing to `lucid`.
   # We also lay down the script for cumuluslinux which is currently a copy of wheezy.
```
Comment (melissa) [inline manifests/setup/cows.pp]: Can you update these comments as well?

**Comment by melissa** [comment]
It looks like you got rid of the squeeze and lucid files. We still need them.

**Comment by melissa** [comment]
pbuilderrc.erb also needs to be updated

File: manifests/setup/cows.pp [hunk 2]
```diff
@@ -26,6 +26,9 @@
     'vivid',
     'wheezy',
     'CumulusLinux-2.2',
+    'wily',
+    'xenial'
```
Comment (melissa) [inline manifests/setup/cows.pp]: we also eventually will need to add stretch to this list

**Comment by stahnma** [comment]
Why do we need the squeeze and lucid files if they're renamed? @melissa I think having the base setup named for distributions that are super old and shouldn't exist anymore is really confusing. (see second commit in PR)

**Comment by melissa** [comment]
I totally agree they shouldn't be the base for the symlinks, but we're still building cows for those distros, so we still need a corresponding file for those distros. They can be symlinks like everything else

**Comment by stahnma** [comment]
@melissa ready for another review.

**Comment by melissa** [comment]
I know I mentioned stretch, but we should probably limit this to just re-factoring the links, and adding wily... Xenial and Stretch will create broken cows because there aren't corresponding repos on apt.puppetlabs.com yet :(

File: manifests/setup/cows.pp (hunk 1 above)
Comment (melissa) [inline manifests/setup/cows.pp]: The comment should still be updated

**Comment by melissa** [comment]
:+1: pending comment update

**Comment by stahnma** [comment]
Which comment?

File: manifests/setup/cows.pp (hunk 1 above)
Comment (melissa) [inline manifests/setup/cows.pp]: @stahnma this comment

File: manifests/setup/cows.pp [hunk 3]
```diff
@@ -118,10 +119,10 @@
 
   # Earlier debian and ubuntu versions won't have the scripts for newers versions for
   # debootstrap. To be safe, we provide scripts for all platforms that we support building
-  # cows of. All Ubuntu varieties are symlinks pointing to `squeeze` and all debian
-  # varieties are symlinks pointing to `lucid`.
+  # cows of. All Debian varieties are symlinks now pointing to `debian` and all Ubuntu
+  # varieties are symlinks pointing to `ubuntu`.
   # We also lay down the script for cumuluslinux which is currently a copy of wheezy.
-  debbuilder::util::file_on_disk { ['CumulusLinux-2.2', 'jessie', 'lucid', 'precise', 'sid', 'squeeze', 'stable', 'stretch', 'testing', 'trusty', 'unstable', 'utopic', 'vivid', 'wheezy']:
+  debbuilder::util::file_on_disk { ['CumulusLinux-2.2', 'jessie', 'lucid', 'precise', 'sid', 'squeeze', 'stable', 'testing', 'trusty', 'unstable', 'utopic', 'vivid', 'wheezy', 'wily' ]:
```
Comment (melissa) [inline manifests/setup/cows.pp]: Yeah, you were right, this is broken. We need to add 'ubuntu' and 'debian' to this list. I just opened a PR for that

**Pull request merged**